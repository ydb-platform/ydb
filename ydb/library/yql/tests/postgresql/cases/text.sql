--
-- TEXT
--
SELECT text 'this is a text string' = text 'this is a text string' AS true;
SELECT text 'this is a text string' = text 'this is a text strin' AS false;
CREATE TABLE TEXT_TBL (f1 text);
INSERT INTO TEXT_TBL VALUES ('doh!');
INSERT INTO TEXT_TBL VALUES ('hi de ho neighbor');
select reverse('abcde');
select i, left('ahoj', i), right('ahoj', i) from generate_series(-5, 5) t(i) order by i;
select quote_literal('');
select quote_literal('abc''');
select quote_literal(e'\\');
/*
 * format
 */
select format(NULL);
select format('Hello');
select format('Hello %%');
select format('Hello %%%%');
select format('Hello %s');
