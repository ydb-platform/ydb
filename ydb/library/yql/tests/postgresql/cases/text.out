--
-- TEXT
--
SELECT text 'this is a text string' = text 'this is a text string' AS true;
 true 
------
 t
(1 row)

SELECT text 'this is a text string' = text 'this is a text strin' AS false;
 false 
-------
 f
(1 row)

CREATE TABLE TEXT_TBL (f1 text);
INSERT INTO TEXT_TBL VALUES ('doh!');
INSERT INTO TEXT_TBL VALUES ('hi de ho neighbor');
select reverse('abcde');
 reverse 
---------
 edcba
(1 row)

select i, left('ahoj', i), right('ahoj', i) from generate_series(-5, 5) t(i) order by i;
 i  | left | right 
----+------+-------
 -5 |      | 
 -4 |      | 
 -3 | a    | j
 -2 | ah   | oj
 -1 | aho  | hoj
  0 |      | 
  1 | a    | j
  2 | ah   | oj
  3 | aho  | hoj
  4 | ahoj | ahoj
  5 | ahoj | ahoj
(11 rows)

select quote_literal('');
 quote_literal 
---------------
 ''
(1 row)

select quote_literal('abc''');
 quote_literal 
---------------
 'abc'''
(1 row)

select quote_literal(e'\\');
 quote_literal 
---------------
 E'\\'
(1 row)

/*
 * format
 */
select format(NULL);
 format 
--------
 
(1 row)

select format('Hello');
 format 
--------
 Hello
(1 row)

select format('Hello %%');
 format  
---------
 Hello %
(1 row)

select format('Hello %%%%');
  format  
----------
 Hello %%
(1 row)

select format('Hello %s');
ERROR:  too few arguments for format()
